cmd/ps3netsrv-go/handler.go

```diff
@@ -22,18 +22,60 @@ func (h *Handler) HandleOpenDir(ctx *server.Context, path string) bool {
 	log := h.Log.With(zap.Stringer("remote", ctx.RemoteAddr), zap.String("path", path))
 	log.Info("Open dir")
 
-	info, err := h.Fs.Stat(path)
+	handle, err := h.Fs.Open(path)
+	if err != nil {
+		log.Warn("Open failed", zap.Error(err))
+		return false
+	}
+
+	info, err := handle.Stat()
 	if err != nil {
 		log.Warn("Stat failed", zap.Error(err))
 		return false
 	}
 
+	if ctx.ROFile != nil {
+		if err := ctx.ROFile.Close(); err != nil {
+			log.Warn("Close already opened ctx.ROFile failed", zap.Error(err))
+		}
+	}
+
+	ctx.ROFile = handle
 	ctx.Cwd = &path
 
 	// it's crucial to send "true" for directory and "false" for file
 	return info.IsDir()
 }
 
+func (h *Handler) HandleReadDirEntry(ctx *server.Context) os.FileInfo {
+	log := h.Log.With(zap.Stringer("remote", ctx.RemoteAddr))
+	log.Info("Read Dir Entry")
+
+	if ctx.Cwd == nil || ctx.ROFile == nil {
+		log.Warn("Reading non-opened dir")
+		return nil
+	}
+
+	info, err := ctx.ROFile.Readdir(1)
+	if errors.Is(err, io.EOF) {
+		return nil
+	} else if err != nil {
+		log.Warn("Readdir failed", zap.Error(err))
+		return nil
+	}
+
+	// Stat resolves a symlink
+	fileInfo, err := h.Fs.Stat(*ctx.Cwd + "/" + info[0].Name())
```

Comment:
`filepath.Join` should be used here to be cross-platform

Reply:
`filepath.Join` is now used.

We don't have a fileInfo now that Readdirnames() is used, so we need the stat call.